packages/headless/src/features/facets/facet-set/facet-set-controller-actions.ts

```diff
@@ -28,18 +28,28 @@ export const executeToggleFacetSelect = createAsyncThunk<
   {
     facetId: string;
     selection: FacetValue;
+    onAfterSend?: (
+      getAnalyticsAction: () => ReturnType<
+        typeof getAnalyticsActionForToggleFacetSelect
+      >
+    ) => void;
   },
   AsyncThunkSearchOptions<FacetSection & ConfigurationSection>
 >(
   'facet/executeToggleSelect',
-  ({facetId, selection}, {dispatch, extra: {validatePayload}}) => {
-    const analyticsAction = getAnalyticsActionForToggleFacetSelect(
+  (
+    {
       facetId,
-      selection
-    );
+      selection,
+      onAfterSend = (getAnalytics) => executeSearch(getAnalytics()),
```

Comment:
I'm not 100% sure about this solution.

This is a somewhat retrocompatible way to send execute a generic search action while still handling analytics.

This way, we can simply use `fetchProductListing` when needed, and discard the analytics event (or not)